controllers/mongodb_users.go

```diff
@@ -0,0 +1,51 @@
+package controllers
+
+import (
+	"fmt"
+
+	mdbv1 "github.com/mongodb/mongodb-kubernetes-operator/api/v1"
+	"github.com/mongodb/mongodb-kubernetes-operator/pkg/kube/secret"
+	apiErrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/types"
+)
+
+func (r ReplicaSetReconciler) ensureUserResources(mdb mdbv1.MongoDBCommunity) error {
```

Comment:
[q] can we add short docstring for every function introduced in this PR? (exported ones especially, but I would add a comment for each function in general)

Reply:
done